# Case Study on Testing of Hypothesis

##### A company started to invest in digital marketing as a new way of their product promotions. For that they collected data and decided to carry out a study on it. 

##### ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

##### ● The company needs to check whether there is any dependency between the features “Region” and “Manager”.

##### The given dataset is 'Sales_add.csv'. Help the company to carry out their study with the help of data provided.

### Importing the modules, reading and analysing the dataset

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
sale_info = pd.read_csv(r'C:\Users\Dell\Downloads\Sales_add.csv')
sale_info.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [3]:
sale_info.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

#### Insights:

It is cleary visible from the above analysis that after adoptiong digital marketing techniques there is an increase accounted in all the metrics related to the given sales data such as mean, minimum and maximum values. The mean sales increased from 149239.954545 to 231123.727273 which is almost almost 55% increase in sales. 

### Check to find whether there is any increase in sales after stepping into digital marketing.

In [4]:
import scipy.stats as stats

tscore, p = stats.ttest_ind(sale_info['Sales_After_digital_add(in $)'],
                               sale_info['Sales_before_digital_add(in $)'],
                               equal_var=True, alternative='greater')
print('The p-value is :', p)

The p-value is : 1.3071840034523225e-16


#### Insights

In this case, we reject the Null Hypothesis and accept the Alternate Hypothesis since the p-value(1.3071840034523225e-16) is considerably lesser than the Alpha value(0.05). Hence, it is clear that there is an increase in sales after adoptiong digital marketing techniques.

### Check to find whether there is any dependency between the features “Region” and “Manager”.

In [6]:
from scipy.stats import chi2_contingency

contigency= pd.crosstab(sale_info['Region'], sale_info['Manager'])
contigency

# Chi-square test of independence
c, p, dof, expected = chi2_contingency(contigency)
print('The p-value is :',p)

The p-value is : 0.5493991051158094


#### Insights:

In this case, we accept the Null Hypothesis and reject the Alternate Hypothesis since the p-value(0.5493991051158094)is greater than the Alpha value(0.05). Hence, it is clear the features “Region” and “Manager” are not depended to each other.